src/hotspot/share/gc/parallel/parallelScavengeHeap.cpp

```diff
@@ -539,6 +539,51 @@ void ParallelScavengeHeap::object_iterate(ObjectClosure* cl) {
   old_gen()->object_iterate(cl);
 }
 
+void ParallelScavengeHeap::object_iterate_parallel(ObjectClosure* cl,
+                                                   HeapBlockClaimer* claimer) {
+  int block_index;
+  // Iterate until all blocks are claimed
+  while (claimer->claim_and_get_block(&block_index)) {
+    if (block_index == HeapBlockClaimer::EdenIndex) {
+      young_gen()->eden_space()->object_iterate(cl);
+    } else if (block_index == HeapBlockClaimer::SurvivorIndex) {
+      young_gen()->from_space()->object_iterate(cl);
+      young_gen()->to_space()->object_iterate(cl);
+    } else {
+      old_gen()->block_iterate(cl, block_index);
```

Comment:
here is a type mismatch between the type of `block_index` and the second parameter (int vs. uint) too